assignment2/src/app/shared/user.model.ts

```diff
@@ -0,0 +1,11 @@
+export class UserModel {
+  constructor(public role: string, public id: number, private token: string, public expToken: Date) {
+  }
+
+  get getTokenUser() {
+    if (!this.token || new Date() >= this.expToken) {
+      return null;
+    }
+    return this.token;
```

Comment:
Please let Backend handle expire date of the token.

Reply:
cái này là khi request API lên server, server trả về JSON có token và expDate trong đó, mục đích là kiểm tra expDate trong JSON trả về có hợp lệ hay ko và sử dụng trong trường hợp user close tab và mở lại thì kiểm tra lần nữa xem expDate có còn thời hạn hay ko

Reply:
Ok. Để khi nào vào project thật em sẽ hiểu anh nói gì.